_posts/2019-04-25-snapshots.md

```diff
@@ -83,6 +83,37 @@ void APP_DISP_lvgl_flush (const lv_color_t* pData, const DISPLAY_BOUNDS_T* const
 }
 ```
 
+- Convert pixel to required format:
+
+```c
+/** Convert lvgl RGB565 pixel to RGB888 pixel
+ *
+ * @note: convert potential gray colors to true gray color.
+ *
+ */
+static uint32_t DISP_IMPL_lvgl_formatPixel (lv_color_t color){
+
+  lv_color32_t data = (lv_color32_t) lv_color_to32(color);
+  uint8_t aux;
+
+  /* Make potential gray color true gray color */
```

Comment:
Just curious, why is adjusting grayscale necessary?

Reply:
Since I use a particular grayscale gradient, if it's not purely gray, some strips can be seen and smooth gradient is lost.

Reply:
@beibean This seems rather application-specific, is it necessary for the blog post?

Reply:
It is as neccessary as convertPixel function itself. Someone may work with LittlevGL with RGB888, or monochrome display. So, of course convertPixel is application-specific and depends on input-pixel (lvgl color format) and output-pixel(imageMagick command, yes the one I mentioned in the post is not the only one).

I added it just like an example as you suggested, if you think some parts are not neccessary for the blog post (like pure gray convert), feel free to change them, no worries 😉

Reply:
@beibean I updated the post to emphasize the function's application specific nature: 8f97855
@embeddedt What do you think?

Reply:
@kisvegabor Perfect!